### PR TITLE
feat(desktop): adjustable font weight and line height (terminal, editor, chat)

### PR DESCRIPTION
**Pull request by Roshvan**
## What

Adds two new appearance controls — **font weight** (100–900) and **line height** (1–3) — to the Terminal and Editor font settings, alongside the existing family/size controls. They persist in `local-db` and apply **live** across all three text surfaces of the desktop app:

| Surface | How it's applied |
|---|---|
| **Terminal** (xterm) — incl. Claude Code & other CLI agents | Native xterm `fontWeight` + `lineHeight` options, on create and on live appearance updates (both the v1 cache and v2 runtime paths) |
| **Editor + diff viewer** (CodeMirror / Pierre) | `fontWeight` on the host; line height computed as `fontSize * multiplier`, replacing the previously hardcoded `*1.5` |
| **Markdown / chat prose** | Body line-height + base font-weight follow the **Editor** settings via CSS variables; headings and bold text keep their own weights/line-heights |

## Why

The only adjustable typography today is family + size. Users want denser or more breathable text and lighter/heavier weights — especially in the terminal, where Claude Code and other agents run.

## Notes

- **No behavior change when unset.** Each surface keeps its existing default (terminal weight 400 / line-height 1.0; editor weight 400 / line-height 1.5; markdown styles keep their tuned defaults). The markdown CSS variables are only set when the user has an explicit override.
- Markdown prose intentionally tracks the **Editor** settings rather than introducing a third settings group; bold/heading semantics are preserved because those elements declare their own weights.
- The diff viewer (Pierre) has no `--diffs-font-weight` variable, so weight is set on the host element — non-bold (inheriting) tokens pick it up while keyword/emphasis token weights from the syntax theme stay intact.
- `local-db` migration `0042_add_font_weight_line_height` adds four nullable columns (generated via `drizzle-kit generate`).

## Testing

- `bun run typecheck` (desktop + local-db) — pass
- `bun run lint` — clean (0)
- `bun test` for `font-settings`, terminal `font-settle` / ws-transport — pass; extended `font-settings.test.ts` with weight/line-height validation + transform cases (32 tests pass)

---

<a href="https://stagereview.app/superset-sh/superset/pull/5229">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://stagereview.app/assets/gh-open-in-stage-dark.svg">
    Open in Stage
  </picture>
</a>

---
## Summary by cubic
Add adjustable font weight (100–900) and line height (1–3) to Terminal, Editor, and chat/markdown. Values persist in `local-db`, apply live across terminal runtime, editor/diff viewer, and markdown, and keep existing defaults when unset.

- **New Features**
  - Terminal: native `fontWeight`/`lineHeight` on create and live updates; defaults 400 / 1.0.
  - Editor + diff viewer: host `fontWeight`; line height = fontSize × multiplier (replaces fixed 1.5); syntax token weights unchanged; diff viewer sets `--diffs-line-height` and host weight.
  - Chat/Markdown: prose follows editor via CSS vars (`--md-font-weight`, `--md-line-height`) with fallbacks; headings/bold unchanged.
  - Settings: new weight/line-height inputs with validation (100–900 int, 1–3), rounds weight to nearest 100 on blur, live preview; tests cover bounds and passthrough.

- **Migration**
  - `0042_add_font_weight_line_height` adds four nullable `settings` columns (terminal/editor weight and line height). No action needed.

<sup>Written for commit 5e659a384029d9f4dcd301fc7813e4a704a68dfc. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/superset-sh/superset/pull/5229?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

## Summary by CodeRabbit

* **New Features**
  * Font weight (100–900) and line-height (1–3) are now configurable for editor, terminal, and markdown, with per-component overrides and persisted settings.
  * Settings UI: numeric inputs, live preview, and reset behavior for weight and line-height.

* **Style**
  * Markdown and theme styles now honor font-weight and line-height via CSS variables.

* **Tests**
  * Added validation and transformation tests covering bounds, fractional/nullable values, and edge cases.

### CODE REVIEW

**Comment by ghost** [comment]
Ready to review this PR? Stage has broken it down into 7 individual chapters for you:

| | Title |
|---|-------|
| 1 | [Add font weight and line height to schema](https://stagereview.app/superset-sh/superset/pull/5229/chapters/1) |
| 2 | [Update font settings API and validation](https://stagereview.app/superset-sh/superset/pull/5229/chapters/2) |
| 3 | [Implement adjustable typography in Terminal](https://stagereview.app/superset-sh/superset/pull/5229/chapters/3) |
| 4 | [Implement adjustable typography in Editor](https://stagereview.app/superset-sh/superset/pull/5229/chapters/4) |
| 5 | [Apply editor typography to Diff Viewer](https://stagereview.app/superset-sh/superset/pull/5229/chapters/5) |
| 6 | [Apply editor typography to Markdown and Chat](https://stagereview.app/superset-sh/superset/pull/5229/chapters/6) |
| 7 | [Add UI controls for font weight and line height](https://stagereview.app/superset-sh/superset/pull/5229/chapters/7) |

<a href="https://stagereview.app/superset-sh/superset/pull/5229">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://stagereview.app/assets/gh-open-in-stage-dark.svg">
    Open in Stage
  </picture>
</a>

<sub>Chapters generated by <a href="https://stagereview.app">Stage</a> for commit 5e659a384029d9f4dcd301fc7813e4a704a68dfc on Jun 11, 2026 6:59pm UTC.</sub>

File: packages/local-db/drizzle/0042_add_font_weight_line_height.sql [hunk 1]
```diff
@@ -0,0 +1,4 @@
+ALTER TABLE `settings` ADD `terminal_font_weight` integer;--> statement-breakpoint
+ALTER TABLE `settings` ADD `terminal_line_height` real;--> statement-breakpoint
+ALTER TABLE `settings` ADD `editor_font_weight` integer;--> statement-breakpoint
+ALTER TABLE `settings` ADD `editor_line_height` real;
```
[the remark this answers is not in the document]
Comment (Roshvan) [reply packages/local-db/drizzle/0042_add_font_weight_line_height.sql]: Fixed — added the trailing newline to the migration file in 088c03e1.

File: apps/desktop/src/renderer/routes/_authenticated/settings/appearance/components/AppearanceSettings/components/FontSettingSection/FontSettingSection.tsx [hunk 2]
```diff
@@ -99,6 +129,34 @@ export function FontSettingSection({ variant }: FontSettingSectionProps) {
...
+				!Number.isNaN(value) &&
+				value >= FONT_WEIGHT_MIN &&
+				value <= FONT_WEIGHT_MAX
+			) {
+				setFontSettings.mutate({ [config.weightKey]: value });
+			}
+		},
+		[setFontSettings, config.weightKey],
+	);
+
```
[the remark this answers is not in the document]
Comment (Roshvan) [reply apps/desktop/src/renderer/routes/_authenticated/settings/appearance/components/AppearanceSettings/components/FontSettingSection/FontSettingSection.tsx:148]: Good catch — fixed in 088c03e1. `handleFontWeightBlur` now snaps to the nearest 100 (and clamps to 100–900), matching the `step=100` UI so off-step values like 350 can't silently clamp to a font's nearest bucket.

**Comment by Roshvan** [comment]
Addressed review feedback in `088c03e1` (amended into the single commit):

- **CodeRabbit (Major) — v1 `CodeEditor.tsx` reconfigure omitted `fontWeight`/`lineHeight`:** confirmed and fixed. The create-once path already had them but the live reconfigure effect didn't, so post-mount editor weight/line-height changes wouldn't apply until remount. Added both to the `createCodeMirrorTheme` call and the effect deps. (The v2 editor already handled this correctly.)
- **greptile (P2) — migration missing trailing newline:** fixed.
- **greptile (P2) — font weight not snapped to step:** fixed; weight now rounds to the nearest 100 and clamps to 100–900 on blur.

**Not addressed — CodeRabbit pre-merge "Docstring coverage 13% < 80%":** declining. This repo doesn't use docstring-coverage as a convention (lint/format is Biome; no JSDoc requirement), and the changed code is self-descriptive. Adding docstrings purely to hit a generic threshold would be noise. Happy to revisit if a maintainer wants specific functions documented.

typecheck + lint clean, `font-settings`/`font-settle` tests pass.

**Comment by Roshvan** [comment]
@avipeltz this is ready for review.

- Rebased onto latest `main` as a single clean commit.
- Scope is v2-only — `screens/main/` (v1) is untouched (`git diff main -- screens/main/` is empty).
- All prior review feedback addressed (editor live-reconfigure, weight snapping to nearest 100, migration trailing newline).
- Local typecheck, lint (biome), and font-settings tests all green; core CI passed on the identical pre-rebase diff.
